common-npm-packages/azure-arm-rest/azCliUtility.ts

```diff
@@ -188,4 +189,46 @@ function isAzVersionGreaterOrEqual(azVersionResultOutput: string, versionToCompa
         tl.error(`Error checking Azure CLI version: ${error.message}`);
         return false;
     }
+}
+
+async function getLatestAzureModuleReleaseVersion(moduleName: string): Promise<string> {
+    try {
+        let request = new webClient.WebRequest();
+        request.uri = `https://api.github.com/repos/Azure/${moduleName}/releases`;
+        request.method = 'GET';
+        request.headers = request.headers || {};
+        const response = await webClient.sendRequest(request);
+        const lastestCliRelease = moduleName === "azure-powershell" ? response?.body?.filter(x => x?.tag_name?.match(/^v\d+\.\d+\.0/))?.[0] : response?.body?.[0];
+        return lastestCliRelease?.tag_name
+    } catch (err) {
+        tl.error(`Error checking Azure version: ${err.message}`);
+    }
+}
+
+export async function validateAzModuleVersion(moduleName: string, CurrentVersion: string, displayName: string, versionsToReduce: number, isMajor: boolean = false): Promise<void> {
+    const DisplayWarningForOlderAzVersion: boolean = tl.getPipelineFeature("Show_Warning_On_old_version");
+    try {
+        if (DisplayWarningForOlderAzVersion) {
+            const latestRelease: string = await getLatestAzureModuleReleaseVersion(moduleName);
+            if (latestRelease) {
+                const [latestsemver, latestMajor, latestMinor] = latestRelease.match(/(\d+).(\d+).(\d+)/);
+                const [currentsemver, currentMajor, currentMinor] = CurrentVersion.match(/(\d+).(\d+).(\d+)/);
+                tl.debug(`the currentsemver Version is ${currentsemver}`);
+                tl.debug(`the latestsemver version is ${latestsemver}`);
+                let displayWarning = false;
+                if (isMajor && (Number(currentMajor) < (Number(latestMajor) - versionsToReduce))) {
```

Comment:
I think there are redundant brackets. It might simplified:
```ts
if (isMajor && Number(currentMajor) < Number(latestMajor) - versionsToReduce)
```

Reply:
Review points addressed as you suggested